wire-ios-request-strategy/Tests/Sources/Request Strategies/User/SelfSupportedProtocolsRequestBuilderTests.swift

```diff
@@ -0,0 +1,189 @@
+
+        // when
+        let request = builder.buildTransportRequest()
+
+        // then
+        XCTAssertEqual(request?.method, .put)
+    }
+
+    func testBuildTransportRequest_thenPayloadIsSet() throws {
+        // given
+        let builder = makeBuilder(supportedProtocols: [.proteus, .mls])
+
+        // when
+        let request = builder.buildTransportRequest()
+
+        // then
+        let payload = try XCTUnwrap(request?.payload as? [String: [String]])
+        let supportedProtocols = payload["supported_protocols"]
+        XCTAssert(supportedProtocols?.contains("proteus") == true)
+        XCTAssert(supportedProtocols?.contains("mls") == true)
+    }
+
+    func testBuildTransportRequest_thenAPIVersionIsSet() {
+        // given
+        let builder = makeBuilder(apiVersion: defaultAPIVersion)
+
+        // when
+        let request = builder.buildTransportRequest()
+
+        // then
+        XCTAssertEqual(request?.apiVersion, defaultAPIVersion.rawValue)
+    }
+
+    // MARK: Upstream Request
+
+    func testBuildUpstreamRequest_givenAPIVersion0_thenDontBuildRequest() {
+        // given
+        let builder = makeBuilder(apiVersion: .v0)
+
+        // when
+        let request = builder.buildUpstreamRequest(keys: .init())
+
+        // then
+        XCTAssertNil(request)
+    }
+
+    func testBuildUpstreamRequest_givenAPIVersion4_thenBuildRequest() {
+        // given
+        let builder = makeBuilder(apiVersion: .v4)
+
+        // when
+        let request = builder.buildUpstreamRequest(keys: .init())
+
+        // then
+        XCTAssertNotNil(request)
+    }
+
+    func testBuildUpstreamRequest_givenAPIVersion5_thenBuildRequest() {
+        // given
+        let builder = makeBuilder(apiVersion: .v5)
+
+        // when
+        let request = builder.buildUpstreamRequest(keys: .init())
+
+        // then
+        XCTAssertNotNil(request)
+    }
+
+    func testBuildUpstreamRequest_thenTransportRequestPathIsSet() {
+        // given
+        let builder = makeBuilder(apiVersion: .v4)
+
+        // when
+        let request = builder.buildUpstreamRequest(keys: .init())
+
+        // then
+        XCTAssertEqual(request?.transportRequest.path, "/v4/self/supported-protocols")
+    }
+
+    func testBuildUpstreamRequest_thenKeysAreSet() {
+        // given
+        let key = "supportedProtocols"
+        let builder = makeBuilder()
+
+        // when
+        let request = builder.buildUpstreamRequest(keys: [key])
+
+        // then
+        XCTAssertEqual(request?.keys, [key])
+    }
+
+    func testBuildUpstreamRequest_thenUserInfo() throws {
+        // given
+        let builder = makeBuilder(supportedProtocols: [.proteus, .mls])
+
+        // when
+        let request = builder.buildUpstreamRequest(keys: .init())
+
+        // then
+        XCTAssert(request?.userInfo.isEmpty == true)
+    }
+
+    // MARK: Helpers
+
+    private func makeBuilder(
+        apiVersion: APIVersion? = nil,
```

Comment:
suggestion (non-blocking): make apiVersion non optional but with default value of the defaultApiVersion

Reply:
Unfortunately it's not possible to reference the var `minimumAPIVersion` as default argument here. That's why I chose this pattern to use `nil` and internally fall back to use the default.